### PR TITLE
InfluxDB installation tutorial

### PR DESCRIPTION
**Pull request by timothy-storageos**

### CODE REVIEW

**Review by actionbuddha — COMMENTED** [review]
Tim, this is looking good, but it needs some embellishment.

On the first page, we present a cluster that is initialising but with no context:

- In this tutorial we will be installing InfluxDB inside Kubernetes using StorageOS volumes. InfluxDB is a time series database that allows us to bla bla and is a great stateful workload to run on StorageOS
- we should be clear that the cluster is partially installed, and that the user needs to pause to wait for initialisation - make the 'I am ready to go' line in the shell very clear - use the green colouring from the self-eval perhaps?
- the bash code used at init time is visible in the scenario - possible to obfuscate in favour of simple status messages echoed?
- can we use the storageclass 'storageos'? If memory serves the operator creates that in addition to 'fast' (to be confirmed by @Arau )
- mention that the use case draws inspiration from our use case in the docs and give link
- where we are deploying manifests, be clear on the git repo and path we are sourcing from - perhaps have the user check them out in a command to avoid needing to keep this scenario in lock step with the use case repo?
- second page 'Provision storage' is more about 'Installing InfluxDB' I think - change of title?

In short, give more context, and make the style a little more narrative - we're taking the user on a journey here from installing InfluxDB through to putting some data in and using it - that should feel exciting and ultimately satisfying.

**Comment by Arau** [comment]
>  can we use the storageclass 'storageos'? If memory serves the operator creates that in addition to 'fast' (to be confirmed by @Arau )

We can set the StorageClass name in the CR when installing StorageOS. Not an addition, but instead. The 

`spec.storageClassName` defines the name of the default class.

**Review by Arau — COMMENTED** [review]
The PR looks good. Any chance @timothy-storageos that you can post the URL that is renders the scenario?

**Comment by timothy-storageos** [comment]
> Tim, this is looking good, but it needs some embellishment.
> 
> On the first page, we present a cluster that is initialising but with no context:
> 
>     * In this tutorial we will be installing InfluxDB inside Kubernetes using StorageOS volumes. InfluxDB is a time series database that allows us to bla bla and is a great stateful workload to run on StorageOS

_Added much more context to the introductory page_
> 
>     * we should be clear that the cluster is partially installed, and that the user needs to pause to wait for initialisation - make the 'I am ready to go' line in the shell very clear - use the green colouring from the self-eval perhaps?

_Removed a lot of extraneous output to make this line much clearer_

> 
>     * the bash code used at init time is visible in the scenario - possible to obfuscate in favour of simple status messages echoed?

_I'm unable to hide the commands run in foreground scripts on katacoda, and if we run the script in the background, we can't send output to the user. I have tried a workaround, which involves clearing the terminal as part of the script, leaving only the "storageos initialising" text. This means the verbose output disappears almost instantly and also helps make it clear when it's ok to start the tutorial._
> 
>     * can we use the storageclass 'storageos'? If memory serves the operator creates that in addition to 'fast' (to be confirmed by @Arau )

_edited the manifest files and tutorial text to reflect this_
> 
>     * mention that the use case draws inspiration from our use case in the docs and give link

_done_ 
> 
>     * where we are deploying manifests, be clear on the git repo and path we are sourcing from - perhaps have the user check them out in a command to avoid needing to keep this scenario in lock step with the use case repo?

_provided a link to the git repository and some commentary clarifying where the files come from
> 
>     * second page 'Provision storage' is more about 'Installing InfluxDB' I think - change of title?

_title changed_ 
> 
> 
> In short, give more context, and make the style a little more narrative - we're taking the user on a journey here from installing InfluxDB through to putting some data in and using it - that should feel exciting and ultimately satisfying.

_tutorial should now be more clear and satisfying_

File: influxdb/000-install-storageos.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  csiControllerExpandPassword: c3RvcmFnZW9z
+END
+
+kubectl create -f- <<END
+apiVersion: "storageos.com/v1"
+kind: StorageOSCluster
+metadata:
+  name: "my-storageos-cluster"
+  namespace: "storageos-operator"
+spec:
```
[the remark this answers is not in the document]
Comment (timothy-storageos) [reply influxdb/000-install-storageos.sh:43]: Thank you

**Comment by timothy-storageos** [comment]
> The PR looks good. Any chance @timothy-storageos that you can post the URL that is renders the scenario?

https://www.katacoda.com/timothy-storageos/scenarios/influxdb

File: influxdb/020-deploy-pods-service.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+The next step is to create an InfluxDB host pod, a service that exposes port
+8086 on that pod, and a client pod for interaction with the host.
+
+Create the service account, service, StatefulSet and client pod from their
+respective yaml files:
+
+`kubectl create -f ./`{{execute}}
```
Comment (Arau) [inline influxdb/020-deploy-pods-service.md:7]: When building an scenario we have to keep in mind the maintainability of the solution. We don't want to have to make the same change in 2 different places. In this case, if the use case needs changing then we would have to apply the changes into the scenario. That is not ideal. 

It is better to do the 
```
git clone ..../use-cases
cd storageos-usecases
```

```
kubectl create -f ./influxdb
```

In this way, people can also related to use the same yamls in their clusters if they want to.

**Review by Arau — CHANGES_REQUESTED** [review]

**Review by avestuk — APPROVED** [review]

**Review by Arau — APPROVED** [review]

**Review by aeroniero33 — APPROVED** [review]
Nice job! LGTM

**Review by actionbuddha — APPROVED** [review]
LGTM - nice!
